lib/Types.js

```diff
@@ -0,0 +1,199 @@
+ *     12s2                  zoned         D myzone  12s 2
+ *     4f2                   float         D myfloat  4f
+ *     8f4                   real/double   D myfloat  8f
+ *     3b                    binary        D mybin   (any)
+*/
+
+/**
+ * @param {number} length - The number of binary characters.
+ * @returns {string} - The binary data type format expected by xml service.
+ */
+
+function Binary(length) {
+  return { type: `${length}b` };
+}
+
+/**
+ * @param {number} length - The number of characters within the varchar.
+ * @returns {string} - The character data type format expected by xml service.
+ */
+function Char(length) {
+  return { type: `${length}a` };
+}
+
+/**
+ * @param {number} length - The number of characters.
+ * @returns {string} - The long varchar data type format expected by
+ * xml service.
+ */
+function LongVarchar(length) {
+  return { type: `${length}a`, varying: '4' };
+}
+
+/**
+ * @param {number} length - The number of characters.
+ * @returns {string} - The varchar data type format expected by xml service.
+ */
+function Varchar(length) {
+  return { type: `${length}a`, varying: '2' };
+}
+
+/**
+ * @returns {string} - The double data type format expected by xml service.
+ */
+function Double() {
+  return { type: '8f4' };
+}
+
+/**
+ * @returns {string} - The float data type format expected by xml service.
+ */
+function Float() {
+  return { type: '4f2' };
+}
+
+/**
+ * @param {number} totalDigits -The number of integer digits.
+ * @param {number} decimalDigits - The number of decimal digits.
+ * @returns {string} - The packed data type format expected by xml service.
+ */
+function PackedDecimal(totalDigits, decimalDigits) {
+  return { type: `${totalDigits}p${decimalDigits}` };
+}
+
+/**
+ * @param {number} totalDigits - The number of integer digits.
+ * @param {number} decimalDigits - The number of decimal digits.
+ * @returns {string} - The zoned data type format expected by xml service.
+ */
+function ZonedDecimal(totalDigits, decimalDigits) {
+  return { type: `${totalDigits}s${decimalDigits}` };
+}
+
+/**
+ * @returns {string} The signed big int (int64) data type format expected by
+ * xml service.
+ */
+function SignedBigInt() {
+  return { type: '20i0' };
+}
+
+/**
+ * @returns {string} The signed byte (int8) data type foramt expected by
+ * xml service.
+ */
+function SignedByte() {
+  return { type: '3i0' };
+}
+
+/**
+ * @returns {string} The signed int (int32) data type format expected by xml service.
+ */
+function SignedInt() {
+  return { type: '10i0' };
+}
+
+/**
+ * @returns {string} The signed short (int16) data type format expected by
+ * xml service.
+ */
+function SignedShort() {
+  return { type: '5i0' };
+}
+
+// aliases for signed types
+const BigInt = SignedBigInt;
+const Byte = SignedByte;
+const Int = SignedInt;
+const Short = SignedShort;
+
+/**
+ * @returns {string} The unsigned big int (int64) data type format expected by
```

Comment:
Should these `@returns` be adjusted? They don't actually return a string.

Reply:
Yup these need to updated to return objects instead.